libvirt/domain.go

```diff
@@ -763,6 +799,10 @@ func setNetworkInterfaces(d *schema.ResourceData, domainDef *libvirtxml.Domain,
 						// have a valid lease and then read the IP we have been assigned, so we can
 						// do the mapping
 						log.Printf("[DEBUG] Do not have an IP for '%s' yet: will wait until DHCP provides one...", hostname)
+						err = network.Ref()
+						if err != nil {
+							return err
+						}
```

Comment:
I prefer to leave the task of freeing the `network` instance to whoever is going to process `partialNetIfaces`.

GitHub doesn't allow me to add comments to the lines above because they are not affected by the change: there are two more `return` statement, we have to introduce a `defer network.Free()` before them.

Reply:
I'm confused by this comment. This code does precisely that: leaves the responsibility of freeing the network to the one processing partialNetIfaces. The use of the `Ref()`, as explained in another comment, is to simplify the logic: always free, but increment the reference only in the case the network is added to `partialNetIfaces`.